core/app/models/spree/order_contents.rb

```diff
@@ -40,6 +40,7 @@ def after_add_or_remove(line_item, options = {})
       shipment.present? ? shipment.update_amounts : order.ensure_updated_shipments
       PromotionHandler::Cart.new(order, line_item).activate
       Adjustable::AdjustmentsUpdater.update(line_item)
+      TaxRate.adjust(order, [line_item]) if line_item.quantity == 1
```

Comment:
A quantity of 1 doesn't necessarily mean line item is being created... Not sure how we can best determine that, but I think we have to pass that around in the options hash.